sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -249,6 +249,7 @@ case class GroupExpression(children: Seq[Expression]) extends Expression {
   override def nullable: Boolean = false
   override def foldable: Boolean = false
   override def dataType: DataType = throw new UnsupportedOperationException
+  def checkInputDataTypes(): TypeCheckResult = TypeCheckResult.TypeCheckSuccess
```

Comment:
I sent another PR so that we can remove this dummy `GroupExpression`, https://github.com/apache/spark/pull/6706